docs/01-app/03-api-reference/03-file-conventions/dynamic-routes.mdx

```diff
@@ -145,6 +145,97 @@ export default async function Page(props: PageProps<'/[locale]'>) {
 - Since the `params` prop is a promise. You must use `async`/`await` or React's use function to access the values.
   - In version 14 and earlier, `params` was a synchronous prop. To help with backwards compatibility, you can still access it synchronously in Next.js 15, but this behavior will be deprecated in the future.
 
+### With Cache Components
+
+When using [Cache Components](/docs/app/getting-started/cache-components) with dynamic routes, params are runtime data that require special handling. During prerendering, param values are unknown (like `/blog/hello-world` or `/blog/123`), so fallback UI using `<Suspense>` boundaries is required when accessing them.
+
+You should use [`generateStaticParams`](/docs/app/api-reference/functions/generate-static-params) to prerender your most popular routes at build time (e.g., `/blog/1`, `/blog/2`, `/blog/3`). This serves two purposes:
+
+- It validates that your route doesn't access dynamic APIs like `cookies()` or `headers()` during prerendering
+- It generates static HTML files for both the specific routes you list and a static shell for runtime params
+
+Prerendered routes load instantly, while runtime params use the static shell to show static content (like headers and navigation) immediately while param-dependent content updates when ready.
+
+The sections below show different patterns, from simplest (all runtime) to most optimized (prerendered samples, and caching). We recommend wrapping param access in Suspense boundaries to provide meaningful loading states.
+
+> **Good to know**: You can check the `X-Nextjs-Cache` response header to verify your caching strategy. It will show `HIT` or `MISS` for disk-cached routes, and won't appear for memory-only caching. Your hosting provider might replace this header with their own version.
+
+#### Without `generateStaticParams`
+
+The simplest approach. All params are runtime data. Param access must be wrapped by Suspense fallback UI. Next.js generates a static shell at build time, and content loads on each request.
+
+```tsx filename="app/blog/[slug]/page.tsx"
+import { Suspense } from 'react'
+
+export default async function Page({ params }: PageProps<'/blog/[slug]'>) {
+  return (
+    <div>
+      <h1>Blog Post</h1>
+      <Suspense fallback={<div>Loading...</div>}>
+        <Content slug={params.then((p) => p.slug)} />
+      </Suspense>
+    </div>
+  )
+}
+
+async function Content({ slug }: { slug: Promise<string> }) {
+  const resolvedSlug = await slug
+  const res = await fetch(`https://api.vercel.app/blog/${resolvedSlug}`)
+  const post = await res.json()
+  return (
+    <article>
+      <h2>{post.title}</h2>
+      <p>{post.content}</p>
+    </article>
+  )
+}
+```
+
+#### With `generateStaticParams`
+
+Prerender your most popular or high-value pages by providing params ahead of time. These pages are generated at build time for instant loading.
```

Comment:
I think one thing we might be missing from this section is that it's ok to prerender all routes ahead of time (depending on how many there are).